cmd/atlas/internal/cmdapi/schema.go

```diff
@@ -20,58 +19,48 @@ import (
 	"github.com/hashicorp/hcl/v2/hclwrite"
 	"github.com/manifoldco/promptui"
 	"github.com/spf13/cobra"
-	"github.com/zclconf/go-cty/cty"
 )
 
-const (
-	urlFlag         = "url"
-	schemaFlag      = "schema"
-	excludeFlag     = "exclude"
-	devURLFlag      = "dev-url"
-	fileFlag        = "file"
-	dsnFlag         = "dsn"
-	varFlag         = "var"
-	autoApproveFlag = "auto-approve"
-)
+func init() {
+	schemaCmd := schemaCmd()
+	schemaCmd.AddCommand(schemaApplyCmd())
+	schemaCmd.AddCommand(schemaCleanCmd())
+	schemaCmd.AddCommand(newDiffCmd())
+	schemaCmd.AddCommand(schemaFmtCmd())
+	schemaCmd.AddCommand(schemaInspectCmd())
+	Root.AddCommand(schemaCmd)
+
+	// Common flags.
+	receivesEnv(schemaCmd)
+}
 
-var (
-	// schemaCmd represents the subcommand 'atlas schema'.
-	schemaCmd = &cobra.Command{
+// schemaCmd represents the subcommand 'atlas schema'.
+func schemaCmd() *cobra.Command {
+	return &cobra.Command{
 		Use:   "schema",
 		Short: "Work with atlas schemas.",
-		Long:  "The `atlas schema` command groups subcommands for working with Atlas schemas.",
-	}
-
-	// SchemaFlags are common flags used by schema commands.
-	SchemaFlags struct {
-		URL     string
-		Schemas []string
-		Exclude []string
-
-		// Deprecated: DSN is an alias for URL.
-		DSN string
-	}
-
-	// ApplyFlags are the flags used in SchemaApply command.
-	ApplyFlags struct {
-		DevURL      string
-		Paths       []string
-		DryRun      bool
-		AutoApprove bool
-	}
-
-	// CleanFlags are the flags used in SchemaClean command.
-	CleanFlags struct {
-		URL         string
-		AutoApprove bool
+		Long:  "The `atlas schema` command groups subcommands working with declarative Atlas schemas.",
 	}
+}
 
-	// SchemaApply represents the 'atlas schema apply' subcommand command.
-	SchemaApply = &cobra.Command{
-		Use:   "apply",
-		Short: "Apply an atlas schema to a target database.",
-		// Use 80-columns as max width.
-		Long: `'atlas schema apply' plans and executes a database migration to bring a given
+// schemaApplyCmd represents the 'atlas schema apply' subcommand command.
+func schemaApplyCmd() *cobra.Command {
+	var (
+		flags = &struct {
+			URL         string   // URL of database to apply the changes on.
+			DevURL      string   // URL of the dev database.
+			Paths       []string // Paths to HCL files.
+			Schemas     []string // Schemas to take into account when diffing.
+			Exclude     []string // List of glob patterns used to filter resources from applying (see schema.InspectOptions).
+			DryRun      bool     // Only show SQL on screen instead of applying it.
+			AutoApprove bool     // Don't prompt for approval before applying SQL.
+			DSN         string   // Deprecated: DSN is an alias for URL.
+		}{}
```

Comment:
```suggestion
		flags struct {
			URL         string   // URL of database to apply the changes on.
			DevURL      string   // URL of the dev database.
			Paths       []string // Paths to HCL files.
			Schemas     []string // Schemas to take into account when diffing.
			Exclude     []string // List of glob patterns used to filter resources from applying (see schema.InspectOptions).
			DryRun      bool     // Only show SQL on screen instead of applying it.
			AutoApprove bool     // Don't prompt for approval before applying SQL.
			DSN         string   // Deprecated: DSN is an alias for URL.
		}
```
Why a pointer is needed?